contractcourt/chain_arbitrator.go

```diff
@@ -627,15 +627,43 @@ func (c *ChainArbitrator) Start() error {
 		c.dispatchBlocks(blockEpoch)
 	}()
 
+	// Create a single read tx that we can use for all of our arbitrators
+	// db lookups as we start them. This reduces our load to a single tx
+	// rather than 2 * number of chan arbitrators.
+	tx, err := c.chanSource.BeginReadTx()
+	if err != nil {
+		if stopErr := c.Stop(); stopErr != nil {
+			log.Errorf("chain arb stop error: %v", err)
+		}
+
+		return fmt.Errorf("could no create chain arb start tx: %v", err)
+	}
+
 	// Finally, we'll launch all the goroutines for each arbitrator so they
 	// can carry out their duties.
 	for _, arbitrator := range c.activeChannels {
-		if err := arbitrator.Start(nil); err != nil {
-			c.Stop()
+		if err := arbitrator.Start(tx); err != nil {
+			if stopErr := c.Stop(); stopErr != nil {
+				log.Errorf("chain arb stop error: %v", err)
+			}
+
+			if rollbackErr := tx.Rollback(); rollbackErr != nil {
+				log.Errorf("chain arb tx rollback failed: %v",
+					err)
+			}
+
 			return err
 		}
 	}
 
+	if err := tx.Rollback(); err != nil {
```

Comment:
This needs to also commit given that each channel arb will/may also update its state on disk.